File: apps/motoboy/app/src/test/kotlin/com/drogsantoantonio/farmadelivery/presentation/ui/screens/RouteStopScheduleLabelsTest.kt
package com.drogsantoantonio.farmadelivery.presentation.ui.screens

import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertNull

class RouteStopScheduleLabelsTest {
  @Test
  fun `formats ISO route stop schedule timestamps for display`() {
    assertEquals("A partir de 15/05 09:30", routeStopScheduleLabel("2026-05-15T12:30:00.000Z"))
  }

  @Test
  fun `falls back to a compact raw timestamp when parsing fails`() {
    assertEquals("A partir de 2026-05-15 12:30", routeStopScheduleLabel("2026-05-15T12:30 sem timezone"))
  }

  @Test
  fun `returns null for blank route stop schedule timestamps`() {
    assertNull(routeStopScheduleLabel("   "))
    assertNull(routeStopScheduleLabel(null))
  }
}

File: apps/motoboy-pwa/src/routeStopScheduleLabels.test.ts
import { describe, it } from "node:test";
import assert from "node:assert/strict";
import { routeStopScheduleLabel } from "./routeStopScheduleLabels";

describe("routeStopScheduleLabel", () => {
  it("formats ISO route stop schedule timestamps for display", () => {
    assert.equal(routeStopScheduleLabel("2026-05-15T12:30:00.000Z"), "A partir de 15/05, 09:30");
  });

  it("falls back to a compact raw timestamp when parsing fails", () => {
    assert.equal(routeStopScheduleLabel("2026-05-15T12:30 sem timezone"), "A partir de 2026-05-15 12:30");
  });

  it("returns null for blank route stop schedule timestamps", () => {
    assert.equal(routeStopScheduleLabel("   "), null);
    assert.equal(routeStopScheduleLabel(null), null);
  });
});
